test/mockwebsocketserver.cpp

```diff
@@ -0,0 +1,61 @@
+#include "mockwebsocketserver.h"
+
+MockWebSocketServer::MockWebSocketServer(quint16 port)
+    : _webSocketServer(new QWebSocketServer(QStringLiteral("Mock Server"), QWebSocketServer::NonSecureMode, this))
+{
+    if (_webSocketServer->listen(QHostAddress::Any, port)) {
+        connect(_webSocketServer, &QWebSocketServer::newConnection, this, &MockWebSocketServer::onNewConnection);
+        connect(_webSocketServer, &QWebSocketServer::closed, this, &MockWebSocketServer::closed);
+        qInfo() << "Open mock websocket server on port:" << port;
+        return;
+    }
+    Q_UNREACHABLE();
+}
+
+MockWebSocketServer::~MockWebSocketServer()
+{
+    qInfo() << "Close mock websocket server";
+    _webSocketServer->close();
+    qDeleteAll(_clients.begin(), _clients.end());
+
+    delete _webSocketServer;
+}
+
+void MockWebSocketServer::sendTextMessage(const QString &message)
+{
+    Q_ASSERT(_clients.size() > 0);
```

Comment:
I guess you want `_clients.size() == 0` ? Otherwise there's no good way to send to the second client.

That being said the original is broadcasting notifications so could be the same here? Send the same message to all the clients at the same time?